crates/cache/src/config.rs

```diff
@@ -579,6 +586,187 @@ impl CacheConfig {
+    /// tolerance limit for these locks. If the time has been changed in the system (i.e. two years
+    /// backwards), the cache system should still work properly. Thus, these locks will be treated
+    /// as expired (assuming the tolerance is not too big).
+    ///
+    /// ### Cache files
+    ///
+    /// Similarly to the locks, the cache files or their metadata might have modification time in
+    /// distant future. The cache system tries to keep these files as long as possible. If the
+    /// limits are not reached, the cache files will not be deleted. Otherwise, they will be
+    /// treated as the oldest files, so they might survive. If the user actually uses the cache
+    /// file, the modification time will be updated.
+    pub fn allowed_clock_drift_for_files_from_future(mut self, drift: Duration) -> Self {
+        self.allowed_clock_drift_for_files_from_future = Some(drift);
+        self
+    }
+
+    /// Soft limit for the file count in the cache directory.
+    ///
+    /// This doesn't include files with metadata. To learn more, please refer to the cache system
+    /// section.
+    pub fn file_count_soft_limit(mut self, limit: u64) -> Self {
+        self.file_count_soft_limit = Some(limit);
+        self
+    }
+
+    /// Soft limit for the total size* of files in the cache directory.
+    ///
+    /// This doesn't include files with metadata. To learn more, please refer to the cache system
+    /// section.
+    ///
+    /// *this is the file size, not the space physically occupied on the disk.
+    pub fn files_total_size_soft_limit(mut self, limit: u64) -> Self {
+        self.files_total_size_soft_limit = Some(limit);
+        self
+    }
+
+    /// If file-count-soft-limit is exceeded and the cache worker performs the cleanup task, then
+    /// the worker will delete some cache files, so after the task, the file count should not
+    /// exceed file-count-soft-limit * file-count-limit-percent-if-deleting.
+    ///
+    /// This doesn't include files with metadata. To learn more, please refer to the cache system
+    /// section.
+    pub fn file_count_limit_percent_if_deleting(mut self, percent: u8) -> Self {
+        self.file_count_limit_percent_if_deleting = Some(percent);
+        self
+    }
+
+    /// If files-total-size-soft-limit is exceeded and cache worker performs the cleanup task, then
+    /// the worker will delete some cache files, so after the task, the files total size should not
+    /// exceed files-total-size-soft-limit * files-total-size-limit-percent-if-deleting.
+    ///
+    /// This doesn't include files with metadata. To learn more, please refer to the cache system
+    /// section.
+    pub fn files_total_size_limit_percent_if_deleting(mut self, percent: u8) -> Self {
+        self.files_total_size_limit_percent_if_deleting = Some(percent);
+        self
+    }
+
+    pub fn build(self) -> Result<CacheConfig> {
```

Comment:
Mind adding some documentation here? I'm thinking specifically about some error cases for example.

Additionally if you're ok with the `&mut self` suggestion above, this would change to either `&self` (ideally) or `&mut self` if necessary.